### ISSUE TITLE
Get commit SHA from branch is wrong

### ISSUE THREAD

**Issue by michael-kokorin** [octokit/octokit.net#1102]
If I use the following code:
```
var commitSha = client.Repository.GetBranch(
				GetSetting(GitHubItSettingKeys.RepositoryOwner),
				GetSetting(GitHubItSettingKeys.RepositoryName),
				branchId)
				.Result.Commit.Sha;

			var sources = client.Repository.Content.GetArchive(
				GetSetting(GitHubItSettingKeys.RepositoryOwner),
				GetSetting(GitHubItSettingKeys.RepositoryName),
				ArchiveFormat.Zipball,
				commitSha)
				.Result;
```

and branchId is different from master then I always get sources from master branch.
This behavior started from version 0.16 and till now. Version 0.15 gets correct result

**Comment by ryangribble** [comment]
Looks like that function inadvertently passes ```String.Empty``` rather than the ```reference``` parameter into the API call 

https://github.com/octokit/octokit.net/blob/master/Octokit/Clients/RepositoryContentsClient.cs#L241

In the meantime you should be able to use the overload that takes an additional ```TimeSpan``` argument (pass in 60 minutes if you want the same behaviour as the overload you are wanting to use), to work around the issue.

**Comment by shiftkey** [comment]
@ryangribble nice spot - yep, that's a bug

**Comment by shiftkey** [comment]
Added the `easy-fix` label as I think this is good for someone new to contribute. I'd also like to see a unit test for this to verify that we're passing through the right value.

**Comment by shiftkey** [comment]
Fixed by #1113
